File: Advanced_hm/src/main/java/ca/six/kjdemo/Temp.java
package ca.six.kjdemo;


class A {

}

class B {
    public static void main(String[] args) {
        A a = new A();
        foo(a);
        boolean isNull = a == null;
        System.out.println(isNull + "; " + a); //=> false;   ca.six.kjdemo.A@2c7b84de
    }

    public static void foo(A arg) {
        System.out.println("B: " + arg); //=> B: ca.six.kjdemo.A@2c7b84de
        arg = null;
    }
}

File: Advanced_hm/src/main/kotlin/ca/six/kjdemo/temp.kt
package ca.six.kjdemo

import io.reactivex.Flowable
import io.reactivex.Observable
import io.reactivex.functions.BiFunction
import io.reactivex.functions.Function
import java.util.concurrent.TimeUnit


fun main() {
    Flowable.interval(1, TimeUnit.SECONDS)

}
